Algorithm.CSharp/BaseBitcoin.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NodaTime;
+using QuantConnect.Brokerages;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+
+    /// <summary>
+    /// Base class for Bitcoin algorithms. This base class provides standard behaviour for trading BTCUSD on Bitfinex
+    /// trading on Bitfinex
+    /// </summary>
+    public abstract class BaseBitcoin : QCAlgorithm
+    {
+
+        const string _bitcoinSymbol = "BTCUSD";
+
+        protected string BitcoinSymbol { get { return _bitcoinSymbol; } }
+
+        public BaseBitcoin()
+        {
+            Portfolio = new BitfinexSecurityPortfolioManager(Securities, Transactions);
+            SetBrokerageModel(BrokerageName.BitfinexBrokerage, AccountType.Margin);
```

Comment:
I think people may run into problems without overriding the vanilla margin calls. Margin funding is cheap so lot of people are fully leveraged for extended periods of time. Having the engine automatically reduce a position due to maxed-out leverage would not be popular.